From 849289703b1c8ef1461d0d0f6a5465e20f3c0932 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 29 Apr 2019 20:32:04 +0200
Subject: [PATCH 1/3] Use CMAKE_INSTALL_LIBDIR for LIB_INSTALL_DIR

Respect the libdir cmake already set up for us instead of using
debian-specific multiarch paths.

Patch was rebased multiple times from 4.0 to 8.0 now.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34b55554..36879db6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -429,15 +429,6 @@ endif ()
 ##############################################################################
 # Installation directories
 
-if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    # Debian multiarch support
-    execute_process(COMMAND dpkg-architecture -qDEB_HOST_MULTIARCH
-        OUTPUT_VARIABLE ARCH_SUBDIR
-        ERROR_QUIET
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-endif()
-
 if (WIN32 OR APPLE)
     # On Windows/MacOSX, applications are usually installed on a directory of
     # their own
@@ -446,12 +437,8 @@ if (WIN32 OR APPLE)
     set (LIB_ARCH_INSTALL_DIR lib)
 else ()
     set (DOC_DEFAULT_INSTALL_DIR share/doc/${CMAKE_PROJECT_NAME})
-    set (LIB_INSTALL_DIR lib${LIB_SUFFIX}/${CMAKE_PROJECT_NAME})
-    if (ARCH_SUBDIR)
-        set (LIB_ARCH_INSTALL_DIR lib/${ARCH_SUBDIR}/${CMAKE_PROJECT_NAME})
-    else ()
-        set (LIB_ARCH_INSTALL_DIR lib${LIB_SUFFIX}/${CMAKE_PROJECT_NAME})
-    endif ()
+    set (LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${CMAKE_PROJECT_NAME})
+    set (LIB_ARCH_INSTALL_DIR ${LIB_INSTALL_DIR})
 endif ()
 
 # Allow customization of the doc installation dir (Slackware uses different
-- 
2.21.0

